realm/realm-library/src/main/java/io/realm/RealmMap.java

```diff
@@ -0,0 +1,667 @@
+     * collected. If the RealmMap is garbage collected, the change listener will stop being
+     * triggered. To avoid this, keep a strong reference for as long as appropriate e.g. in a class
+     * variable.
+     * <p>
+     * <pre>
+     * {@code
+     * public class MyActivity extends Activity {
+     *
+     *     private RealmMap<String, Dog> dogs; // Strong reference to keep listeners alive
+     *
+     *     \@Override
+     *     protected void onCreate(Bundle savedInstanceState) {
+     *       super.onCreate(savedInstanceState);
+     *       dogs = realm.where(Person.class).findFirst().getDogs();
+     *       dogs.addChangeListener(new RealmChangeListener<RealmMap<String, Dog>>() {
+     *           \@Override
+     *           public void onChange(RealmMap<String, Dog> map) {
+     *               // React to change
+     *           }
+     *       });
+     *     }
+     * }
+     * }
+     * </pre>
+     *
+     * @param listener the change listener to be notified.
+     * @throws IllegalArgumentException if the change listener is {@code null}.
+     * @throws IllegalStateException    if you try to add a listener from a non-Looper or
+     *                                  {@link android.app.IntentService} thread.
+     * @see io.realm.RealmChangeListener
+     */
+    public void addChangeListener(RealmChangeListener<RealmMap<K, V>> listener) {
+        mapStrategy.addChangeListener(this, listener);
+    }
+
+    /**
+     * Removes the specified change listener.
+     *
+     * @param listener the change listener to be removed.
+     * @throws IllegalArgumentException if the change listener is {@code null}.
+     * @throws IllegalStateException    if you try to remove a listener from a non-Looper Thread.
+     */
+    public void removeChangeListener(MapChangeListener<K, V> listener) {
+        mapStrategy.removeChangeListener(this, listener);
+    }
+
+    /**
+     * Removes the specified change listener.
+     *
+     * @param listener the change listener to be removed.
+     * @throws IllegalArgumentException if the change listener is {@code null}.
+     * @throws IllegalStateException    if you try to remove a listener from a non-Looper Thread.
+     * @see io.realm.RealmChangeListener
+     */
+    public void removeChangeListener(RealmChangeListener<RealmMap<K, V>> listener) {
+        mapStrategy.removeChangeListener(this, listener);
+    }
+
+    /**
+     * Removes all user-defined change listeners.
+     *
+     * @throws IllegalStateException if you try to remove listeners from a non-Looper Thread.
+     * @see io.realm.RealmChangeListener
+     */
+    public void removeAllChangeListeners() {
+        mapStrategy.removeAllChangeListeners();
+    }
+
+    /**
+     * Indicates whether a map has any listeners attached to it.
+     *
+     * @return {@code true} if any listeners have been added, {@code false} otherwise.
+     */
+    public boolean hasListeners() {
```

Comment:
I don't think we have this method anywhere else in the API? What is the use case for it?

Reply:
Because if we add it here, we should probably also add it to all the other observable classes for consistency sakes.

Reply:
That's true. I think I added it only due to testing purposes because I wanted to assert that there were actually no attached listeners after calling `removeListener`.

I would rather not make any changes to the other observable collections, so I can either make this method package-private or remove it altogether.

Reply:
I leave it as a package-private method.